### DISCUSSION TITLE
How to use with vscode extension like crates.io

### DISCUSSION THREAD

**Discussion by rain2307** [kellnr/kellnr#470] [Q&A] [answered]
I deployed it locally and want to use it with the vscode plugin. I used BarbossHack.crates-io, which can suggest the latest version, but when I configure the private indexServerURL, it returns a 502 error. My address is: http://192.xxxxx:3000/. I wonder if it's a problem with the plugin itself or if it's incompatible. Has anyone encountered this?

**Comment by secana** [comment, marked as answer]
Hi @rain2307. I do not use vscode, so I don't know what's going on. The API for kellnr is the same as the API for crates.io, else cargo would not work. I don't know if the plugin needs more than kellnr currently provides. The index of kellnr is can be found under http(s)://yourkellnrhostname/api/v1/crates/ so maybe give that a try? It seems that the plugin supports alternative registries: https://github.com/BarbossHack/crates-io/blob/master/src/core/AlternateRegistry.ts

**Reply by rain2307** [reply]
Thanks for your reply.
I tried `http://192.168.8.16:18000/api/v1/crates/` but it shows a empty webpage. 
The extension default value of `crates.indexServerURL` is `https://index.crates.io/`. 

[crates.io](https://github.com/BarbossHack/crates-io/tree/master)

**Reply by rain2307** [reply]
I will check the extensions code. thank you.

**Comment by rain2307** [comment]
https://github.com/citreae535/sparse-crates

I  use this one now, it works perfectly
